src/impl/tpls/KokkosBlas1_iamax_tpl_spec_decl.hpp

```diff
@@ -467,9 +490,9 @@ namespace Impl {
         int idx;                                                              \
         KokkosBlas::Impl::CudaBlasSingleton& s =                              \
             KokkosBlas::Impl::CudaBlasSingleton::singleton();                 \
-        cublasIzamax(s.handle, N,                                             \
-                     reinterpret_cast<const cuDoubleComplex*>(X.data()), LDX, \
-                     &idx);                                                   \
+        KOKKOS_CUBLAS_SAFE_CALL_IMPL(cublasIzamax(                            \
```

Comment:
@e10harvey The default is the scalar values are passed by reference on the host. But you are right. If ```CUBLAS_POINTER_MODE_DEVICE ``` was set by user code, this might be wrong. So, I have added pointer mode checking here.